core/server/worker/src/main/java/alluxio/worker/grpc/DoraWorkerClientServiceHandler.java

```diff
@@ -109,14 +113,19 @@ public StreamObserver<ReadRequest> readBlock(StreamObserver<ReadResponse> respon
   public void getStatus(GetStatusPRequest request,
                         StreamObserver<GetStatusPResponse> responseObserver) {
     try {
-      String ufsFilePath = request.getPath();
-      UfsFileStatus status = mUfsFileStatusCache.get(request.getPath());
+      String alluxioFilePath = request.getPath();
+
+      String ufsFullPath = PathUtils.concatPath(mRootUFS, alluxioFilePath);
+      String fn = new AlluxioURI(alluxioFilePath).getName();
+
+      UfsFileStatus status = mUfsFileStatusCache.get(ufsFullPath);
       GetStatusPResponse response = GetStatusPResponse.newBuilder()
           .setFileInfo(
               FileInfo.newBuilder()
-                  .setName(status.getName())
-                  .setPath(ufsFilePath)
-                  .setUfsPath(ufsFilePath)
+                  .setFileId(ufsFullPath.hashCode())
```

Comment:
Looks like the bug fix is not related to whether we use relative path or absolute path. always using full path can reduce many of the conversion overhead and support caching for multiple systems, what's the bottlenecks/other bugs i missing which results in current dora does not support full URL?

Reply:
The problems or fixes here are:
1. Name = A file name without SEPARATE.
2. Path = Alluxio visible path, a.k.a the path in Alluxio name space.
3. UfsPath = Full path in UFS. This ufsPath will be used later by "open()" to open the actually file as Stream.


It should be clear that: Any path/file names from Alluxio client API are Alluxio based file/path name. They are in Alluxio namespace. They must be transformed to UFS file/path name when accessing them in UFS.
This is done via Mount Tables.
On the other hand, any file/path name returned in FileStatus from UFS, must be stripped, removing the leading UFS root path.

This is the main problem here.
I don't see other bugs or bottlenecks.

Reply:
1. Name = A file name without SEPARATE. fully agree
2. Path. Alluxio name space path <-- this is the part may need discussion today
3. UfsPath <-- fully agree

If we use relative path in Alluxio, multiple UFSes will have the same relative path in Alluxio namespace, and may result in conflicts?

Since we do not have master now, can we always keep full Path, probably do not need two fields, Path == UfsPath == full ufs path